qiskit/providers/aer/jobs/aerjob.py

```diff
@@ -24,13 +24,21 @@
 
 
 class AerJob(Job):
-    """AerJob class.
-
-    Attributes:
-        _executor (futures.Executor): executor to handle asynchronous jobs
-    """
+    """AerJob class for Qiskit Aer Simulators."""
 
     def __init__(self, backend, job_id, fn, qobj, executor=None):
+        """ Initializes the asynchronous job.
+
+        Args:
+            backend(AerBackend): the backend used to run the job.
+            job_id(str): a unique id in the context of the backend used to run the job.
+            fn(function): a callable function to execute qobj on backend.
+                :class:`AerBackend` sets :meth:`AerBackend._run()` function that takes
+                qobj(QasmQobj) and job_id(string) as an argument.
```

Comment:
I'm not quite sure what the new description means exactly, sorry, but I'm also not familiar enough with Aer to really suggest a proper alternative.  I think a lot of my confusion is that I don't quite understand the separation of responsibilities between `AerBackend` and `AerJob` here; in the whole library, `AerJob` only seems to be instantiated within `AerBackend`, so `fn` _cannot_ be anything but `backend._run`.  Is `AerJob` meant to be instantiable by users/other things?  If not, then I'm not really sure why the `fn` argument exists at all (`AerJob` could just do it internally); it seems to be very very dependent on `AerBackend` agreeing with the `executor` argument.

That said, this really isn't the point of the PR, so I'm not 100% sure what I'm suggesting here.  Mostly I think I'm just saying that I can't suggest better documentation because I don't understand what's happening!

I _think_ we're saying this:
```suggestion
            fn(function): a callable function to execute qobj on backend.
                This should usually be a bound :meth:`AerBackend._run()` method,
                with the signature `(qobj: QasmQobj, job_id: str) -> Result`.
```
?

Reply:
Thank you for the update! I think it is better than my description and correct. 

> Is `AerJob` meant to be instantiable by users/other things? 

No. this instance is created by only `AerBackend` as you mentioned. 

> If not, then I'm not really sure why the `fn` argument exists at all (`AerJob` could just do it internally); it seems to be very very dependent on `AerBackend` agreeing with the `executor` argument.

That is certainly true.  AerJob can work well without `fn` argument. If this argument is removed, AerJob will become simpler.  This argument exists for some time so I just use it without thinking.